modules/test/playwright/tests/roles-admin-web/main/roles.spec.ts

```diff
@@ -1941,3 +1942,136 @@ test(
+
+		const user = await apiHelpers.headlessAdminUser.postUserAccount();
+
+		userData[user.alternateName] = {
+			name: user.givenName,
+			password: 'test',
+			surname: user.familyName,
+		};
+
+		const companyId = await page.evaluate(() => {
+			return Liferay.ThemeDisplay.getCompanyId();
+		});
+
+		const role = await apiHelpers.headlessAdminUser.postRole({
+			name: getRandomString(),
+			rolePermissions: [
+				{
+					actionIds: ['VIEW_CONTROL_PANEL'],
+					primaryKey: companyId,
+					resourceName: '90',
+					scope: 1,
+				},
+				{
+					actionIds: ['ACCESS_IN_CONTROL_PANEL'],
+					primaryKey: companyId,
+					resourceName:
+						'com_liferay_roles_admin_web_portlet_RolesAdminPortlet',
+					scope: 1,
+				},
+				{
+					actionIds: ['VIEW'],
+					primaryKey: companyId,
+					resourceName: 'com.liferay.portal.kernel.model.Role',
+					scope: 1,
+				},
+				{
+					actionIds: ['UPDATE'],
+					primaryKey: companyId,
+					resourceName: 'com.liferay.portal.kernel.model.Role',
+					scope: 1,
+				},
+			],
+			roleType: 'regular',
+		});
+
+		await apiHelpers.headlessAdminUser.assignUserToRole(
+			role.externalReferenceCode,
+			user.id
+		);
+
+		await performLogout(page);
+		await performLoginViaApi({page, screenName: user.alternateName});
+
+		await rolesPage.goto(false);
+
+		await rolesPage.rolesTable.changeView('Table');
+
+		const guestRoleName = 'Guest';
+
+		await rolesPage.rolesTable.search(guestRoleName);
+
+		await expect(rolesPage.rolesTable.cell(guestRoleName)).toBeVisible();
+
+		await (await rolesPage.rolesTable.rowActions(guestRoleName)).click();
+
+		await expect(rolesPage.duplicateMenuItem).not.toBeVisible();
+
+		await performLogout(page);
+		await performLoginViaApi({page, screenName: 'test'});
+
+		await rolesPage.goto();
+
+		await rolesPage.rolesTable.search(guestRoleName);
+
+		await expect(rolesPage.rolesTable.cell(guestRoleName)).toBeVisible();
+
+		await (await rolesPage.rolesTable.rowActions(guestRoleName)).click();
+		await rolesPage.duplicateMenuItem.click();
+
+		await expect(rolesPage.duplicateFrameNewRoleNameInput).toBeVisible();
+
+		await rolesPage.duplicateFrameNewRoleNameInput.fill(guestRoleName);
+		await rolesPage.duplicateFrameSaveButton.click();
+
+		await expect(rolesPage.duplicateFrameErrorMessage).toBeVisible();
+
+		await rolesPage.duplicateFrameNewRoleNameInput.clear();
+		await rolesPage.duplicateFrameSaveButton.click();
+
+		await expect(rolesPage.duplicateFrameEmptyErrorMessage).toBeVisible();
+
+		const duplicateRoleName = 'role' + getRandomInt();
+
+		await rolesPage.duplicateFrameNewRoleNameInput.fill(duplicateRoleName);
+		await rolesPage.duplicateFrameSaveButton.click();
+
+		try {
+			await expect(rolePage.keyInput).toHaveValue(duplicateRoleName);
+
+			await rolePage.definePermissionsLink.click();
+
+			await expect(
+				roleDefinePermissionsPage.noRoleMessage
+			).not.toBeVisible();
+		}
+		finally {
+			await rolesPage.goto();
```

Comment:
if you do the other change remember here to logout and login again with the test user